cli/cmd/contracts.go

```diff
@@ -94,6 +95,84 @@ var (
 		},
 	}
 
+	contractsStorageCmd = &cobra.Command{
+		Use:   "storage",
+		Short: "WebAssembly smart contracts storage operations",
+	}
+
+	contractsStorageDumpCmd = &cobra.Command{
+		Use:   "dump <instance-id>",
+		Short: "Dump instantiated public contract state in JSON",
+		Args:  cobra.ExactArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			cfg := cliConfig.Global()
+			npa := common.GetNPASelection(cfg)
+			strInstanceID := args[0]
+
+			if npa.ParaTime == nil {
+				cobra.CheckErr("no paratimes configured")
+			}
+
+			instanceID, err := strconv.ParseUint(strInstanceID, 10, 64)
+			cobra.CheckErr(err)
+
+			ctx := context.Background()
+			conn, err := connection.Connect(ctx, npa.Network)
+			cobra.CheckErr(err)
+
+			res, err := conn.Runtime(npa.ParaTime).Contracts.InstanceRawStorage(ctx, client.RoundLatest, contracts.InstanceID(instanceID))
```

Comment:
Add support for pagination as the entire storage can be arbitrarily large.

Reply:
Hm, this will be tricky, because we can only paginate based on the number of items in the storage. Now the smart contract can always have a huge map stored as a single item in the storage and we wouldn't be able to handle that?

Reply:
The size of single items is limited.

Let's just focus on supporting pagination of the "many items in storage" case.